### PR TITLE
gcc@13: add Modula 2 support

### PR DESCRIPTION
**Pull request by schaecsn**
Enable support for Modula 2 and add a test case.

Deparallelization of 'make install' is necessary, at least under macOS, otherwise the make command is stuck for reasons unknown.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by fxcoudert** [comment]
Hi, I don't think we want to add features to versioned (legacy) formulas. How about adding it to the main GCC formula?

**Comment by schaecsn** [comment]
> Hi, I don't think we want to add features to versioned (legacy) formulas. How about adding it to the main GCC formula?

Modula 2 came in with gcc 13. I wanted to enable it in both, gcc 13 and 14. I happened to raise a PR for gcc 13 first.

I'm going to raise a PR for gcc 14 soon.

File: Formula/g/gcc@13.rb [hunk 1]
```diff
@@ -35,6 +36,7 @@ class GccAT13 < Formula
 
   on_linux do
     depends_on "binutils"
+    depends_on "flex"
```
Comment (fxcoudert) [inline Formula/g/gcc@13.rb:39]: Does this use Apple's `flex` on macOS? If so, it should be `uses_from_macos` instead.

File: Formula/g/gcc@13.rb [hunk 2]
```diff
@@ -120,7 +122,8 @@ def install
       # To make sure GCC does not record cellar paths, we configure it with
       # opt_prefix as the prefix. Then we use DESTDIR to install into a
       # temporary location, then move into the cellar path.
-      system "make", install_target, "DESTDIR=#{Pathname.pwd}/../instdir"
+      # Deparallelize for modula 2 as make install will block indefinitely.
+      ENV.deparallelize { system "make", install_target, "DESTDIR=#{Pathname.pwd}/../instdir" }
```
Comment (fxcoudert) [inline Formula/g/gcc@13.rb:126]: Has this been reported upstream?

Comment (schaecsn) [reply Formula/g/gcc@13.rb:127]: Modula 2 requires flex for building. Flex comes with MacOS, but not with the Ubuntu CI container.

Your "instead" puzzles me. Don't you mean "as well"? I would assume that this is the way to go (and that is what I just pushed):

```
  uses_from_macos "flex"

  on_linux do
    ...
    depends_on "flex"
  end
```

File: Formula/g/gcc@13.rb (hunk 1 above)
Comment (schaecsn) [reply Formula/g/gcc@13.rb:40]: Modula 2 requires flex. Flex comes with MacOS, but not with the Ubuntu CI container.

Your "instead" puzzles me. Don't you mean "as well"? I would assume that this is the way to go (and that is what I just pushed):

```
  uses_from_macos "flex"

  on_linux do
    ...
    depends_on "flex"
  end
```

**Comment by schaecsn** [comment]
The docker instructions on https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md#homebrew-docker-container created a docker image with only 4 GB of memory where my brew of gcc@13 runs out of memory. I don't know why the CI docker image on github does not run out of memory.

What made linux CI pass was

```
  on_linux do
    ...
    depends_on "flex"
  end
```

`flex` should be just a build-time dependency and declaring it as such

```
  on_linux do
    depends_on "flex" => :build
  end
```

makes the build fail in configure with

```
  checking for suffix of executables... flex: fatal internal error, exec of /tmp/gccA13-20240704-6531-uaccr6/gcc-13.3.0/missing m4 failed
```

Why do I need all in a sudden `m4` as a build-time dependency? Let me try that next ...

File: Formula/g/gcc@13.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/g/gcc@13.rb]: No, `uses_from_macos` means: "On macOS use the one from system, and on other platforms install this formula"

**Comment by schaecsn** [comment]
Without an explicit dependency on `m4,` I see it getting fetched on the failed linux CI run

```
==> Determining dependencies...
==> brew fetch --formulae --retry autoconf automake berkeley-db@5 binutils bison bzip2 ca-certificates cmake expat flex gdbm gettext gmp gpatch help2man icu4c isl krb5 libedit libffi libmpc libnsl libtirpc libtool libxcrypt libxml2 lz4 lzip m4 mpdecimal mpfr ncurses openssl@3 perl pkg-config python-setuptools python@3.11 python@3.12 readline sqlite unzip xz zlib zstd
```


but not really installed, at least the `gcc` configure phase does not find it

```
  checking for gm4... no
  checking for gnum4... no
  checking for m4... no
```

When adding `m4` as a build dependency, linux CI succeeds. Anyway, I have now these two build dependencies and linux CI is happy

```
  uses_from_macos "flex" => :build
  uses_from_macos "m4" => :build
```

**Comment by fxcoudert** [comment]
Support was added to the main GCC formula at https://github.com/Homebrew/homebrew-core/pull/177143

At this stage, unless there is a very compelling reason why we should add it to legacy formula, we will decline this PR. Thanks for getting the ball rolling, though!

**Pull request closed without merging**